contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,553 @@
+
+    ```yaml
+    apiVersion: v1
+    kind: LimitRange
+    metadata:
+      name: mylimits
+    spec:
+       - default:
+         storage.kubernetes.io/logs: 200Mi
+         storage.kubernetes.io/overlay: 200Mi
+         type: Container
+       - default:
+         storage.kubernetes.io: 1Gi
+         type: EmptyDir
+    ```
+
+3. The limit range will update the pod specification as follows:
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+     - name: fooc
+       resources:
+         limits:
+           storage.kubernetes.io/logs: 200Mi
+           storage.kubernetes.io/overlay: 200Mi
+     volumes:
+     - name: myEmptyDir
+       emptyDir:
+	     resources:
+           limits:
+		     storage.kubernetes.io: 1Gi
+    ```
+
+4. Bob’s “foo” pod can use upto “200Mi” for its containers logs and writable layer each, and “1Gi” for its “myEmptyDir” volume. 
+5. If Bob’s pod “foo” exceeds the “default” storage limits and gets evicted, then Bob can set a minimum storage requirement for his containers and a higher “capacity” for his EmptyDir volumes.
+
+  ```yaml
+  apiVersion: v1
+  kind: pod
+  metadata:
+   name: foo
+  spec:
+   containers:
+   - name: fooc
+     resources:
+     requests:
+	   storage.kubernetes.io/logs: 500Mi
+	   storage.kubernetes.io/overlay: 500Mi
+   volumes:
+   - name: myEmptyDir
+     emptyDir:
+	     resources:
+           limits:
+             storage.kubernetes.io/logs: 2Gi
+  ```
+
+6. It is recommended to require `limits` to be specified for `storage` in all pods. `storage` will not affect the `QoS` Class of a pod since no SLA is intended to be provided for storage capacity isolation. it is recommended to use Persistent Durable Volumes as much as possible and avoid primary partitions.
+
+### Alice manages a Database which needs access to “durable” and fast scratch space
+
+1. Cluster administrator provisions machines with local SSDs and brings up the cluster
+2. When a new node instance starts up, an addon DaemonSet discovers local “secondary” partitions which are mounted at a well known location and creates Local PVs for them if one doesn’t exist already. The PVs will include a path to the secondary device mount points, and a new node annotation that ties the volume to a specific node.  Storage classes and labels may also be specified.  The volume consumes the entire partition.
+
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: local-pv-1
+      annotations:
+        volume.kubernetes.io/node: node-1
+        volume.beta.kubernetes.io/storage-class: local-fast
+    spec:
+      capacity:
+        storage: 100Gi
+      local:
+        path: /var/lib/kubelet/storage-partitions/local-pv-1
```

Comment:
why is the path in the spec, i am assuming the pv gets populated with the path of where the storage is allocated. Putting it in spec, suggests its an input, when its probably just an output, right @msau42 ?